<a href="https://colab.research.google.com/github/MAR-Jemim/python_mini_course/blob/main/Classes/04.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Class 04: Taking user input
---

### What we'll learn today

- ✔️ How to take inputs from a user.
- ✔️ Type conversion/casting

In the last class, we have learned about variables. We have calculated some **Scientifitisitc Mathbalistic** staffs in the **Assignment-03** (hopefully 😶‍🌫).
In thoese calculations, _we_ have entered value. But will we create programs only for ourselves? Nope. That's why we need to take other users' input. The `input()` function is used for this.

So what to do? We will take **input**, store it into a **variable**, after calculating we **print** the result.

Like this:

In [1]:
num = input() # input 4
num *= 5      # *= remember me?
print(num)    # what will it show?

44444


Oops! I thought I gave `4` as input 🤔. Then it is multiplied by `5`. And it should have printed `20` instead of `44444`.

> **Can you say why?** When does this kind of thing happen?

Hmm, when we multiply a string, it gets repeated that many times.

What do we get from here? `input` always takes everything as a string. Fine you do not need to believe me blindly, see yourself.

In [2]:
num = input()
print(num - 3)

TypeError: unsupported operand type(s) for -: 'str' and 'int'

### What to do now?

We can convert the type of input into anything. This is called Type Conversion/Casting.

At first check this out.

In [7]:
num = '3.1415926535897932384626433832'
print(type(num), '\t', num)

num = float(num)
print(type(num), '', num)

num = int(num)
print(type(num), '\t', num)

num = bool(num)
print(type(num), '\t', num)

<class 'str'> 	 3.1415926535897932384626433832
<class 'float'>  3.141592653589793
<class 'int'> 	 3
<class 'bool'> 	 True


See that, we can convert if they are allowed.

There are several conversion function.
- int()
- float()
- str()
- bool()
- set()
- etc...

But some conversions are not allowed. Like `"3.1416"` [str type with radix point] » `3` [int type]

In [8]:
print( int("3.1416") )

ValueError: invalid literal for int() with base 10: '3.1416'

Test yourself what is possible and what is not.

> "With great program, comes great input handling."

<br>

powered by `//>_<` and [`proBOT`](https://codeforces.com/profile/probot001)